controllers/azuremachine_reconciler_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"testing"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api-provider-azure/api/v1alpha2"
+	azure "sigs.k8s.io/cluster-api-provider-azure/cloud"
+	"sigs.k8s.io/cluster-api-provider-azure/cloud/scope"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func TestGetControlPlaneMachines(t *testing.T) {
+	controlPlaneMachine := clusterv1.Machine{
+		ObjectMeta: v1.ObjectMeta{
+			Name:   "control-plane-0",
+			Labels: map[string]string{},
+		},
+	}
+	controlPlaneMachine.ObjectMeta.Labels[clusterv1.MachineControlPlaneLabelName] = "cp-0"
+
+	agentMachine := clusterv1.Machine{
+		ObjectMeta: v1.ObjectMeta{
+			Name:   "machine-0",
+			Labels: map[string]string{},
+		},
+	}
+
+	cases := []struct {
```

Comment:
could you please add a test case for multiple control plane machines as well

Reply:
done